### PR TITLE
chore: move from tap to node:test

### PR DESCRIPTION
**Pull request by eugleenyc**
These changes are related to the issue https://github.com/fastify/fastify/issues/5555
Changed signer.test.js and cookie.test.js to use node:test instead of tap.

#### Checklist

- [x] run `npm run test`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by gurgunday — APPROVED** [review]
LGTM

File: test/cookie.test.js
Comment (dancastillo) [inline test/cookie.test.js:1]: Can you add back in `t.plan`. The test will end before assertions are compelete i believe
Then for the tests with callbacks update to
```js
test('test name here', (t, done) => {
  // set up and test code
  done() // call after last assertion
})

**Review by dancastillo — COMMENTED** [review]
Can you use [test context](https://nodejs.org/api/test.html#class-testcontext) so `assert.strictEqual` -> `t.assert.strictEqual`

**Review by jsumners — CHANGES_REQUESTED** [review]
Please address @dancastillo's concern.

**Comment by Uzlopak** [comment]
@jsumners 
@dancastillo 

Fixed the tests... PTAL.

**Review by jsumners — APPROVED** [review]
Looks good to me.

**Review by gurgunday — APPROVED** [review]

**Pull request merged**